Frontend/android-native/mobility-common/src/main/java/in/juspay/mobility/common/MobilityCommonBridge.java

```diff
@@ -431,6 +445,49 @@ private void getLastKnownLocationFromClientFallback(String callback, boolean ani
     public void initiateLocationServiceClient() {
         if (!isLocationPermissionEnabled()) return;
         resolvableLocationSettingsReq();
+        setKeyboardVisibilityListener();
+    }
+    
+    // Alternate way to get the keyboard status since there is no Keyboard Listener or Events in android
+    private void setKeyboardVisibilityListener() {
+        if (bridgeComponents.getActivity() != null){
+            try {
+                final View parentView = ((ViewGroup) bridgeComponents.getActivity().findViewById(android.R.id.content)).getRootView();
+                onKeyboardVisibilityListener = visible -> {
+                    try {
+                        String encoded = Base64.encodeToString("{}".getBytes(), Base64.NO_WRAP);
+                        String command = String.format("window[\"onEvent'\"]('%s',atob('%s'))", visible ? "onKeyboardOpen" : "onKeyboardClose", encoded);
+                        bridgeComponents.getJsCallback().addJsToWebView(command);
+                    } catch (Exception e) {
+                        Log.e("MobilityCommonBridge", "Error in onVisibilityChanged " + e);
+                    }
+                };
+                onGlobalLayoutListener = new ViewTreeObserver.OnGlobalLayoutListener() {
+
+                    private boolean alreadyOpen;
+                    private final Rect rect = new Rect();
+
+                    @Override
+                    public void onGlobalLayout() {
+                        int defaultKeyboardHeightDP = 100;
+                        int estimatedKeyboardDP = defaultKeyboardHeightDP + 48;
+                        int estimatedKeyboardHeight = (int) TypedValue.applyDimension(TypedValue.COMPLEX_UNIT_DIP, estimatedKeyboardDP, parentView.getResources().getDisplayMetrics());
+                        parentView.getWindowVisibleDisplayFrame(rect);
+                        int heightDiff = parentView.getRootView().getHeight() - (rect.bottom - rect.top);
+                        boolean isShown = heightDiff >= estimatedKeyboardHeight;
+
+                        if (isShown == alreadyOpen) {
+                            return;
+                        }
+                        alreadyOpen = isShown;
+                        onKeyboardVisibilityListener.onVisibilityChanged(isShown);
```

Comment:
Can't we call simple callback here?
`                        String encoded = Base64.encodeToString("{}".getBytes(), Base64.NO_WRAP);
                        String command = String.format("window[\"onEvent'\"]('%s',atob('%s'))", visible ? "onKeyboardOpen" : "onKeyboardClose", encoded);
                        bridgeComponents.getJsCallback().addJsToWebView(command);`
                        
 What is the use of interface ?